src/group_theory/submonoid/basic.lean

```diff
@@ -71,6 +67,28 @@ export zero_mem_class (zero_mem)
 
 attribute [to_additive] one_mem_class
 
+namespace one_mem_class
+
+variables {A M} [has_one M] [set_like A M] [hA : one_mem_class A M] (S' : A)
+include hA
+
+/-- Subobjects of a type with one have a one themselves. -/
+@[to_additive "Subobjects of a type with zero have a zero themselves.",
+priority 900] -- lower priority so other instances are found first
+instance has_one : has_one S' := ⟨⟨1 , one_mem S'⟩⟩
```

Comment:
I think this might make more sense here:

https://github.com/leanprover-community/mathlib/blob/9d37f5063a199f089d5d7c414fb41e510ec8fc68/src/group_theory/submonoid/operations.lean#L393-L409

At any rate we certainly don't need both.

Reply:
shoot, I'm really sorry. That's the second time I've done this too you recently (i.e., adding something we already have). I checked and couldn't find it, but I'll try to be better about that.

I'll just weaken the type class assumptions here to `one_mem_class`.